src/core/server/server.h

```diff
@@ -156,6 +156,14 @@ class Server : public ServerInterface,
     // done. As such, implementations of `LogicalConnection` should cancel the
     // handshake on `Orphan` if still in progress, but not close down the
     // transport.
+    // Implementations are responsible for informing ListenerState about the
+    // following stages of a connection -
+    // 1) Invoke AddLogicalConnection() on accepting a new connection. Do not
+    // invoke if the connection is going to be closed immediately.
+    // 2) Invoke OnHandshakeDone() (irrespective of error) once handshake is
```

Comment:
Would it make sense to say that `OnHandshakeDone()` does not need to be called if `RemoveLogicalConnection()` is called?  It seems like we could easily catch this case in `RemoveLogicalConnection()`, and it might make things a bit easier for transport implementations.

Reply:
good point, added

Reply:
If we're changing this, then we need to change the implementation to match.  Right now, `RemoveLogicalConnection()` is a no-op if there's no config fetcher, because it assumes `OnHandshakeDone()` has already been called.  I think it instead needs to remove the connection unconditionally.

Reply:
right, thanks!